appstore/subscription.go

```diff
@@ -0,0 +1,85 @@
+package appstore
+
+// https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/StoreKitGuide/Chapters/Subscriptions.html#//apple_ref/doc/uid/TP40008267-CH7-SW16
+type NotificationType string
+
+const (
+	// Initial purchase of the subscription.
+	NotificationTypeInitialBuy NotificationType = "INITIAL_BUY"
+	// Subscription was canceled by Apple customer support.
+	NotificationTypeCancel NotificationType = "CANCEL"
+	// Automatic renewal was successful for an expired subscription.
+	NotificationTypeRenewal NotificationType = "RENEWAL"
+	// Customer renewed a subscription interactively after it lapsed.
+	NotificationTypeInteractiveRenewal NotificationType = "INTERACTIVE_RENEWAL"
+	// Customer changed the plan that takes affect at the next subscription renewal. Current active plan is not affected.
+	NotificationTypeDidChangeRenewalPreference NotificationType = "DID_CHANGE_RENEWAL_PREFERENCE"
+)
+
+type NotificationEnv string
+
+const (
+	NotificationEnvSandbox    NotificationEnv = "SANDBOX"
+	NotificationEnvProduction NotificationEnv = "PROD"
+)
+
+type NotificationExpiresDate struct {
+	ExpiresDateMS  string `json:"expires_date"`
+	ExpiresDateUTC string `json:"expires_date_formatted"`
+	ExpiresDatePST string `json:"expires_date_formatted_pst"`
+}
+
+type NotificationReceipt struct {
```

Comment:
Could you tell me the document of this receipt fields?
Is it different from the normal receipt fields?
https://github.com/dogenzaka/go-iap/blob/4b4785955acc4a9b09760e11bd5734a8463300f5/appstore/model.go#L76-L89

Reply:
Have no documents...
It's different with the normal receipt fields.
It's based on the notification body on production environment.